static-site/src/components/Groups/edit-logo-form.jsx

```diff
@@ -0,0 +1,138 @@
+import React, { useEffect, useRef, useState } from "react";
+import { MediumSpacer } from "../../layouts/generalComponents";
+import * as splashStyles from "../splash/styles";
+import { AvatarWithoutMargins, CenteredForm, InputButton } from "./styles";
+
+
+const EditLogoForm = ({ groupName, createErrorMessage, clearErrorMessage }) => {
+  const [ logo, setLogo ] = useState({ current: null, new: null });
+  const [ deletionInProgress, setDeletionInProgress ] = useState(false);
+  const [ uploadInProgress, setUploadInProgress ] = useState(false);
+  // Ref to allow us to create click event for hidden file input
+  const fileInput = useRef(null);
+
+  useEffect(() => {
+    const setCurrentLogo = async () => {
+      const currentLogo = await getGroupLogo();
+      if (!cleanUp) setLogo({ ...logo, current: currentLogo });
+    }
+
+    let cleanUp = false;
+    setCurrentLogo();
+    return () => cleanUp = true;
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const getGroupLogo = async () => {
+    clearErrorMessage();
+    try {
+      const response = await fetch(`/groups/${groupName}/settings/logo`);
+      if (response.status === 404) return null;
+      if (response.ok) return URL.createObjectURL(await response.blob());
+      createErrorMessage(response.statusText);
+    } catch (err) {
+      console.error(err.message)
+      createErrorMessage();
+    }
+  }
```

Comment:
I did this initially (not including `encodeURIComponent` 👀 ), but I ran into browser caching issue that prevented the image from updating even though it was properly updated in S3. 

Is there a better way around the caching than adding a "random" string to the `src` (see [Stackoverflow answer](https://stackoverflow.com/a/126831))?

Reply:
Appending a nonce to the query part to the URL is the easiest, simplest, and most robust way. It's tried and true. Rather than random, it can be a timestamp.

Alternatively, we could serve an appropriate Cache-Control header for the logo to force the browser to revalidate its cache (e.g. `Cache-Control: no-cache`). (We could do it conditionally too, if we wanted, rather than for every request, though at that point I'm not sure it's better than what you already have. :upside_down_face: )

Reply:
Oh right, I also fetch the logo separately to be able to better distinguish 404 errors from other errors. This allows the conditional rendering of "Current logo" + "Delete current logo" button, "No current logo", or the error message banner. 

I don't think I can get the status code from the img `onerror` event...

Reply:
Ah, ok. :+1:

Reply:
(If we don't want to assume that all `onerror` is equivalent to "No current logo", then yeah, I guess we need to keep the manual fetch.)

Reply:
Would still be good to `encodeURIComponent()` these group names in the URLs, though.

Reply:
Encoded group names in URLs in 50565e17c0e8000382deb5d4dc54fb18c6d7c4f1

Reply:
Thanks! Kinda makes me want to make

```js
await fetch(url`/groups/${group.name}/settings`)
```

work.

Reply:
https://gist.github.com/tsibley/4922b5fde8d8b7010b3271782173617d maybe